Userland/Libraries/LibCpp/AST.h

```diff
@@ -281,6 +281,33 @@ class Pointer : public Type {
     RefPtr<Type> m_pointee;
 };
 
+class Reference : public Type {
+public:
+    virtual ~Reference() override = default;
+    virtual const char* class_name() const override { return "Reference"; }
+    virtual void dump(FILE* = stdout, size_t indent = 0) const override;
+    virtual String to_string() const override;
+
+    enum class Kind {
+        Lvalue,
+        Rvalue,
+    };
+
+    Reference(ASTNode* parent, Optional<Position> start, Optional<Position> end, const String& filename, Kind kind)
+        : Type(parent, start, end, filename)
+        , m_kind(kind)
+    {
+    }
+
+    const Type* pointee() const { return m_pointee.ptr(); }
+    void set_pointee(RefPtr<Type>&& pointee) { m_pointee = move(pointee); }
+    Kind kind() const { return m_kind; }
+
+private:
+    RefPtr<Type> m_pointee;
+    Kind m_kind;
```

Comment:
Since it's only set in the c'tor, can be made const.

Reply:
I assume you mean `RefPtr<Type const>`?

Reply:
I was actually referring to `m_kind`.

Reply:
oh
eh, I don't think making that `const` makes any difference, it's just a private enum member